src/util/replace_expr.cpp

```diff
@@ -6,41 +6,44 @@ Author: Daniel Kroening, kroening@kroening.com
 
 \*******************************************************************/
 
-
 #include "replace_expr.h"
+#include "expr_iterator.h"
 
 bool replace_expr(const exprt &what, const exprt &by, exprt &dest)
 {
-  if(dest==what)
+  bool no_change = true;
+
+  for(auto it = dest.depth_begin(), itend = dest.depth_end(); it != itend;)
```

Comment:
Most other places in the code base add a comment `// no ++it` to make it clear that has not been forgotten.